python/test/unit/spacewalk/satellite_tools/test_appstreams.py

```diff
@@ -0,0 +1,163 @@
+"""Unit tests for AppStreams reposync module."""
+
+import os
+import pytest
+from mock import Mock
+
+import spacewalk.satellite_tools.appstreams
+from spacewalk.satellite_tools.appstreams import ModuleMdImporter, Nsvca, Nevra
+
+import gi
+
+# pylint: disable=wrong-import-position
+gi.require_version("Modulemd", "2.0")
+from gi.repository import Modulemd
+
+# pylint: enable=wrong-import-position
+
+
+class TestModuleMdImporter:
+    """Unit tests for ModuleMdImporter."""
+
+    def setup_method(self):
+        self.appstreams = spacewalk.satellite_tools.appstreams
+        self.appstreams.rhnSQL.initDB = Mock()
+        self.appstreams.rhnSQL.commit = Mock()
+
+        module_path = os.path.abspath(__file__)
+        module_directory = os.path.dirname(module_path)
+        modules_yaml_path = os.path.join(module_directory, "sample-modules.yaml")
+        self.importer = ModuleMdImporter(1, modules_yaml_path)
+
+    @pytest.fixture
+    def make_stream(self):
+        def _make_stream(name: str, stream: str, version: int, context: str, arch: str):
+            stream = Modulemd.ModuleStreamV2.new(name, stream)
+            stream.set_version(version)
+            stream.set_context(context)
+            stream.set_arch(arch)
+            return stream
+
+        return _make_stream
+
+    def test_nsvca_repr(self, make_stream):
+        stream = make_stream("my-module", "1.0", 1001, "fedcba98", "x86_64")
+        nsvca = Nsvca(stream)
+        assert "my-module:1.0:1001:fedcba98:x86_64" == str(nsvca)
+
+    @pytest.mark.parametrize(
+        "nevra,expected_output",
+        [
+            (
+                Nevra("my-package", "1", "1.0", "1+MODULE+el9", "x86_64"),
+                "my-package-1:1.0-1+MODULE+el9.x86_64",
+            ),
+            (
+                Nevra("my-package", "0", "1.0", "1+MODULE+el9", "noarch"),
+                "my-package-0:1.0-1+MODULE+el9.noarch",
+            ),
+            (
+                Nevra("my-package", None, "1.0", "1+MODULE+el9", "src"),
+                "my-package-1.0-1+MODULE+el9.src",
+            ),
+        ],
+    )
+    def test_nevra_repr(self, nevra, expected_output):
+        assert expected_output == str(nevra)
+
+    def test_validate(self):
+        self.importer.validate()
+
+        self.importer.modulemd_file = "INVALID_FILE"
+        with pytest.raises(gi.repository.GLib.GError) as exc:
+            self.importer.validate()
+
+        assert exc.value.domain == "modulemd-yaml-error-quark"
+
+    # pylint: disable=protected-access
```

Comment:
[nitpick] This looks like something has the wrong visibility

Reply:
This case tests the protected `_get_modules` method. Black doesn't let this without the `disable` directive.